### DISCUSSION TITLE
given modifier for singleton enum cases

### DISCUSSION THREAD

**Discussion by bishabosha** [scala/scala3#17869] [Feature Requests]
Instead of

```scala
enum Tag[T]:
  case IntTag    extends Tag[Int]
  case StringTag extends Tag[String]

object Tag:
  given Tag[Int] = IntTag
  given Tag[String] = StringTag
```

allow

```scala
enum Tag[T]:
  case given IntTag    extends Tag[Int]
  case given StringTag extends Tag[String]
```

**Comment by LPTK** [comment]
That seems useful, but I'd rather spell it `given case` than `case given`.

**Comment by bishabosha** [comment]
> That seems useful, but I'd rather spell it `given case` than `case given`.

I think that would probably be best - I put it that way because it looked similar to the pattern match syntax
